test/e2e/common/lifecycle_hook.go

```diff
@@ -89,7 +89,7 @@ var _ = framework.KubeDescribe("Container Lifecycle Hook", func() {
 			Testname: Pod Lifecycle, post start exec hook
 			Description: When a post start handler is specified in the container lifecycle using a ‘Exec’ action, then the handler MUST be invoked after the start of the container. A server pod is created that will serve http requests, create a second pod with a container lifecycle specifying a post start that invokes the server pod using ExecAction to validate that the post start is executed.
 		*/
-		It("should execute poststart exec hook properly [NodeConformance]", func() {
+		framework.ConformanceIt("should execute poststart exec hook properly [NodeConformance]", func() {
```

Comment:
Success is verified by looking for the GET in the logs of the serving pod/container?

What image/command is running in the pod/container with the pre-start hook? We need to ensure the primary container isn't racing with the hook.

Is the Netexec image built with multi-arch support?

Note: Some test(s) needs to explicitly validate that pod networking works as intended. Something like this could potentially do that if we constrained the 2 pods to run on different nodes and verified that their IPs weren't the same as the node IPs.

Reply:
a) There are 2 pods present in the scenario:
1) serving pod
- [image](https://console.cloud.google.com/gcr/images/kubernetes-e2e-test-images/GLOBAL/netexec@sha256:738c96036e6e0a90fdae3499c8c9ee9a7d75c0dae70cd22440f69d027c6686d7/details?tab=info) has references to different archs (amd64 / arm / arm64 / ppc64le / s390x)
2) invoker pod
- has the post-start-hook, that will invoke the uri exposed on the "serving pod".
- [image](https://console.cloud.google.com/gcr/images/kubernetes-e2e-test-images/GLOBAL/hostexec@sha256:90dfe59da029f9e536385037bc64e86cd3d6e55bae613ddbe69e554d79b0639d/details?tab=info) has references to different archs (amd64 / arm64/ ppc64le)

b) _Success is verified by looking for the GET in the logs of the serving pod/container?_
 its checking that the log in the invoker pod (that has the hook) contains the consequence of curl invocation.

c) _What image/command is running in the pod/container with the pre-start hook? We need to ensure the primary container isn't racing with the hook._
-  the command run is curl http://" + targetIP + ":8080/echo?msg=poststart

d) _Note_
How would you like us to proceed regarding networking?

e) Each test run creates its own "serving pod", setting it up in the beforeEach section, making sure its created beforehand its usage by creating it with createSync). All tests use the same pod name ("pod-handle-http-request"). If the tests were run in parallel, wouldn't this situation be a problem?

Reply:
Thanks. I see that the image waits forever:
https://github.com/kubernetes/kubernetes/blob/master/test/images/hostexec/Dockerfile

Networking doesn't need to be addressed by these tests. Let's address that in tests focused on that issue.